site/tests/app/libraries/UtilsTester.php

```diff
@@ -430,6 +430,25 @@ public function testGetAutoFillDataDuplicateIds() {
         $this->assertEquals($expected, Utils::getAutoFillData($users));
     }
 
+    public function testConvertBooleans(): void {
+        $values = [
+            'true' => true,
+            'on' => true,
+            'false' => false,
+            'random_string' => false,
+            true => true,
+            false => false,
+            123 => true,
+            1 => true,
+            0 => false
+        ];
```

Comment:
It would be better to use a [Data Provider](https://docs.phpunit.de/en/10.5/writing-tests-for-phpunit.html#data-providers) for this.